### PR TITLE
Update split method definition

### PR DESCRIPTION
**Pull request by sigmaSd**

### CODE REVIEW

**Comment by ringabout** [comment]
Wrong branch. PRs should target to the devel branch.

File: lib/js/jsre.nim [hunk 1]
```diff
@@ -33,7 +33,7 @@ func compile*(self: RegExp; pattern: cstring; flags: cstring) {.importjs: "#.com
 func replace*(pattern: cstring; self: RegExp; replacement: cstring): cstring {.importjs: "#.replace(#, #)".}
   ## Returns a new string with some or all matches of a pattern replaced by given replacement
 
-func split*(pattern: cstring; self: RegExp): seq[cstring] {.importjs: "#.split(#)".}
+func split*(pattern: cstring; self: RegExp | cstring; limit: int): seq[cstring] {.importjs: "#.split(@)".}
```
Comment (ghost) [inline lib/js/jsre.nim:36]: Apart from making the PR to the `devel` branch the signature is wrong - as in https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/split `limit` is an optional argument.

Comment (sigmaSd) [reply lib/js/jsre.nim:36]: I looked  how the codebase handled optional and it seems to just do the above for example flags here https://github.com/nim-lang/Nim/blob/version-1-6/lib/js/jsre.nim#L30 

Is there another way?

Comment (sigmaSd) [reply lib/js/jsre.nim:36]: I'm gonna guess nim doesn't handle js options its either you require them in the definition or omit them.

**Pull request closed without merging**